xdebug_stack.c

```diff
@@ -1383,6 +1402,20 @@ function_stack_entry *xdebug_add_stack_frame(zend_execute_data *zdata, zend_op_a
 	return tmp;
 }
 
+void xdebug_remove_stack_frame(function_stack_entry *fse)
+{
+	size_t i = 0;
+	/* Unlocks the garbage collection of stack frame (function call arguments) objects
+	 * So these objects can be properly GCed by PHP later on */
+	for (i = 0; i < fse->gc_locked_objects_count; i++) {
+#if PHP_VERSION_ID >= 70300
+		GC_DELREF(fse->gc_locked_objects[i]);
+#else
+		GC_REFCOUNT(fse->gc_locked_objects[i])--;
+#endif
+	}
```

Comment:
I'm OK with leaving the function name as it is, because it makes sense to have it as counter to `xdebug_add_stack_frame` as you mentioned. I would probably add a new line in between line 1407 and 1408 though :-).

I would probably also set `fse->gc_locked_objects_count` to `0` at the end here. And shouldn't `fse->gc_locked_objects` be freed? It's allocated at https://github.com/xdebug/xdebug/pull/474/files#diff-70352c82b3e971e39da6aad79610c901R1306

Reply:
Done, but please check I've freed it properly: my C skills are pretty old... :/

Reply:
After reading https://www.owasp.org/index.php/Doubly_freeing_memory I think I should also set `fse->gc_locked_objects = NULL`? Can you confirm/infirm this suggestion @derickr ?